fieldbehavior/fieldbehavior_test.go

```diff
@@ -100,6 +100,17 @@ func TestValidateRequiredFieldsWithMask(t *testing.T) {
 			"missing required field: shipment.origin_site",
 		)
 	})
+	t.Run("missing optional field annotated with required", func(t *testing.T) {
+		t.Parallel()
+		assert.Error(
+			t,
+			ValidateRequiredFieldsWithMask(
+				&examplefreightv1.Site{},
+				&fieldmaskpb.FieldMask{Paths: []string{"personnel_count"}},
+			),
+			"missing required field: personnel_count",
+		)
+	})
```

Comment:
I am running this test without the above changes (other than enabling optional support) and it seems to pass for me.
Can you describe what kind of problem you ran into that prompted this PR? Is there a way to reproduce the issue in a test?

Reply:
My bad, that test wasn't enough. I've added a happy path test that hopefully clears things up a bit.

The gist of the problem is that if a field is annotated with REQUIRED, it can't be set to its default value, `0` in the case of an `int64`. The optional keyword allows for that by differentiating unset from default values. This is reflected in the generated code by making the fields pointers:

```
int64 personnel_count = 1;
```

would generate

```
PersonnelCount int64 `protobuf:"varint,7,opt,name=personnel_count,json=personnelCount,proto3,oneof" json:"personnel_count,omitempty"`
```

whereas

```
int64 personnel_count = 1;
```

would generate

```
PersonnelCount *int64 `protobuf:"varint,7,opt,name=personnel_count,json=personnelCount,proto3,oneof" json:"personnel_count,omitempty"`
```

Therefore, if a field uses the proto3 keyword `optional` but has a `REQUIRED` annotation, it needs to be validated as `!= nil`, which allows for an `int64` to be `0` in a request.